packages/cli/cli/src/commands/upgrade/upgradeGenerator.ts

```diff
@@ -81,22 +83,22 @@ export async function loadAndUpdateGenerators({
             const generatorName = generator.get("name") as string;
             const normalizedGeneratorName = generatorsYml.getGeneratorNameOrThrow(generatorName, context);
 
-            const currentVersion = generator.get("version") as string;
+            const currentGeneratorVersion = generator.get("version") as string;
 
-            const latestVersion = await generatorsYml.getLatestGeneratorVersion(normalizedGeneratorName, context);
-            context.logger.debug(`${generatorName}, ${currentVersion}, ${latestVersion}`);
+            const latestVersion = await generatorsYml.getLatestGeneratorVersion({
+                generatorName: normalizedGeneratorName,
+                cliVersion,
+                currentGeneratorVersion,
+                includeRc,
+                includeMajor,
+                context
+            });
 
             if (latestVersion == null) {
                 continue;
             }
-
-            // check if the versions have the same major version, if not fail, if yes do the upgrade
-            const currentMajor = semver.major(currentVersion);
-            const latestMajor = semver.major(latestVersion);
-            if (currentMajor === latestMajor || includeMajor) {
-                generator.set("version", latestVersion);
-                context.logger.info(chalk.green(`${generatorName} has been upgraded to latest in group: ${groupName}`));
-            }
+            context.logger.debug(`Upgrading ${generatorName} from ${currentGeneratorVersion} to ${latestVersion}`);
```

Comment:
`context.logger.info(chalk.green(`${generatorName} has been upgraded to latest in group: ${groupName}`));`

Let's keep the chalk.green ?

Reply:
sg! not sure why I removed it, likely a copy paste from elsewhere